### PR TITLE
feat(cocoapods): Support for vendored Pods

### PR DESCRIPTION
**Pull request by Dahlgren**
Still need to do some testing on our iOS repos

Most logic is borrowed from the gomod manager

### CODE REVIEW

File: lib/manager/cocoapods/artifacts.ts [hunk 1]
```diff
@@ -78,12 +79,37 @@ export async function updateArtifacts({
   }
   logger.debug(`Returning updated lockfile: ${lockFileName}`);
   const lockFileContent = await readLocalFile(lockFileName);
-  return [
+  const res = [
```
Comment (JamieMagee) [inline lib/manager/cocoapods/artifacts.ts:82]:
```suggestion
  const res: UpdateArtifactsResult[] = [
```

File: lib/manager/cocoapods/artifacts.ts [hunk 2]
```diff
@@ -78,12 +79,37 @@ export async function updateArtifacts({
...
     {
       file: {
         name: lockFileName,
         contents: lockFileContent,
       },
     },
   ];
+
+  const podsDir = join(dirname(packageFileName), 'Pods/');
```
Comment (JamieMagee) [inline lib/manager/cocoapods/artifacts.ts:91]: This isn't crossplatform. I know it's unlikely that someone will be running Renovate on Windows, to update dependencies on an iOS project, but better safe than sorry. 

```suggestion
  const podsDir = join(dirname(packageFileName), 'Pods');
```

**Comment by JamieMagee** [comment]
Looks fine, just needs some test coverage. Thanks!

File: lib/manager/cocoapods/artifacts.ts (hunk 2 above)
Comment (Dahlgren) [reply lib/manager/cocoapods/artifacts.ts:91]: I guess the gomod code should have the same fix as well

Comment (JamieMagee) [reply lib/manager/cocoapods/artifacts.ts:91]: Yeah 🤦. Don't worry, I'll do that in another PR.

**Comment by Dahlgren** [comment]
Added test with added, modified and deleted files in Pods directory

**Review by viceice — APPROVED** [review]

**Pull request merged**